source/AndroidResolver/src/PlayServicesResolver.cs

```diff
@@ -2191,15 +2203,51 @@ public static void SettingsDialog()
+                    versionComparer.Compare("3.4", version) >= 0 ?
                     "implementation" : "compile";
                 if (includeDependenciesBlock) lines.Add("dependencies {");
-                foreach (var packageSpecAndSources in GetPackageSpecs(dependencies: dependencies)) {
-                    lines.Add(String.Format(
-                        "    {0} '{1}' // {2}", includeStatement, packageSpecAndSources.Key,
-                        packageSpecAndSources.Value));
+                // Build a map of dependencies with the max version of that dependency.
+                // If different packages have different versions of the same dependency,
+                // we want to activate only the highest version but still include the
+                // other versions as commented out dependency lines.
+                var dependenciesMaxVersions = new Dictionary<string, string>();
+                string dependencyMaxVersion;
+                foreach( var dependency in dependencies) {
+                    if (dependenciesMaxVersions.TryGetValue(dependency.VersionlessKey, out dependencyMaxVersion)){
+                        if(versionComparer.Compare(dependency.Version, dependencyMaxVersion) < 0) {
+                            dependenciesMaxVersions[dependency.VersionlessKey] = dependency.Version;
+                        }
+                    }
+                    else {
+                        dependenciesMaxVersions[dependency.VersionlessKey] = dependency.Version;
+                    }
+                }
+                foreach (var packageSpecAndSourcesWithVersions in GetPackageSpecsWithVersions(dependencies: dependencies)){
+                    var packageSpecString = packageSpecAndSourcesWithVersions.Key;
+                    var packageSourcesString = packageSpecAndSourcesWithVersions.Value[0];
+                    var packageVersionlessKey = packageSpecAndSourcesWithVersions.Value[1];
+                    var packageVersion = packageSpecAndSourcesWithVersions.Value[2];
+
+                    string line = "    ";
+                    // If this is not the highest version of this dependency, add a line
+                    // but comment it out by adding leading slashes.
+                    if(dependenciesMaxVersions[packageVersionlessKey] != packageVersion) {
+                        PlayServicesResolver.Log(
+                            String.Format(
+                                "Ignoring duplicate package {0} with older version.",
+                                 packageSpecString),
+                                 level: LogLevel.Info
+                            );
+                        line += "// ";
+                    }
+
+                    line += String.Format(
+                            "{0} '{1}' // {2}", includeStatement, packageSpecString,
+                            packageSourcesString);
+                    lines.Add(line);
```

Comment:
Alright,  I think you are over-complicated logic and introduce new untested APIs here.

Here is my proposal:
`GetPackageSpecs()` returns a map from "packageSpec" to "sources"
What you need is to know which "packageSpec" should be commented.
What if you can just construct a Set of packageSpec-To-Comment. This logic should be pretty easy to integrate into your logic to find the max version..
Then you just need to test if the packageSpec is in that set.  If so, add "//". If not, don't add "//".

In this case, you do not need `GetPackageSpecsWithVersions()` nor `DependenciesToPackageSpecsWithVersions()`

Reply:
If you are making this change, ignore all my other comments

Reply:
Done! Thanks for the input :). Makes much more sense to do it this way.